File: squares-of-a-sorted-array/squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        vector<int> v; map<int,int> m;
        for(int i=0;i<nums.size();i++)
        {
            int temp=abs(nums[i]);
            m[temp]++;
        }
        auto it=m.begin();
        while(it!=m.end())
        {
            for(int i=0;i<it->second;i++)
            {
                v.push_back((it->first*it->first));
            }
            it++;
        }
        return v;
    }
};

File: squares-of-a-sorted-array/squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int[] result=new int[nums.length];
        int left=0; int right=nums.length-1;
        for(int i=nums.length-1;i>=0;i--)
        {
            if(Math.abs(nums[left])>nums[right]){
                result[i]=nums[left]*nums[left];
                left++;
            }
            else{
                result[i]=nums[right]*nums[right];
                right--;
            }
        }
        return result;
    }
}